internal/initwd/module_install.go

```diff
@@ -254,6 +270,32 @@ func (i *ModuleInstaller) moduleInstallWalker(ctx context.Context, manifest mods
 					}
 
 					log.Printf("[TRACE] ModuleInstaller: Module installer: %s %s already installed in %s", key, record.Version, record.Dir)
+
+					// Checking for module deprecations in the case no new module versions need installation
+					if addr, isRegistryModule := req.SourceAddr.(addrs.ModuleSourceRegistry); isRegistryModule {
+						regClient := i.reg
+
+						regsrcAddr := regsrc.ModuleFromRegistryPackageAddr(addr.Package)
+						resp, err := regClient.ModuleVersions(ctx, regsrcAddr)
+						if err != nil {
+							log.Printf("[DEBUG] Deprecation for %s could not be checked: call to registry failed: %v", addr.Package.Namespace, err)
```

Comment:
I don't think Namespace is enough to identify this module. Was thinking `addr.Package.String()`

Reply:
I like that errors here will not impede init from taking place. But also this feels like a higher level of log than DEBUG. Can we promote this to [WARN] ?

Reply:
Good point now that I think about it– included 👍  https://github.com/hashicorp/terraform/pull/35231/commits/e3fc6d2d30c24cd5ceff57d8b3bb6fb23cadebe5